posts/2021-06-11-request-timing-21006.adoc

```diff
@@ -0,0 +1,193 @@
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17177[Failed to locate DataSource, null Resourcefactory]
++
+A defect was found in the situation where a customer has deployed a user feature that references a relational database and is also running applications that use global transactions in a configuration where the transaction recovery logs are stored in a database. 
++
+Note that this is not applicable in the default case where transaction recovery logs are stored in a filesystem.
++
+The issue occurred intermittently at server startup and arose when transaction recovery was attempted before data source initialization has been completed. In this situation the server would report `InternalLogExceptions` and failed to start.
+The fix introduces additional coordination between transaction recovery and data source initialization to allow the server to start successfully.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/17040[Revision to httpOption maxKeepAliveRequest default value]
++
+By default, the Liberty HTTP channel configuration closes an incoming `HTTP keep alive connection` after 100 requests (`maxKeepAliveRequests`). The WAS Liberty performance documentation recommends disabling this without qualification and recommends using unlimited requests per persistent connection. Note that there is still the persistent timeout of 30 seconds (`persistTimeout`) which closes an idle `keepalive` socket after 30 seconds.
++
+To fix this issue it was decided to change the default of the `maxKeepAliveRequests` option to `-1`. 
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/16743[Pull in MyFaces 2.3.9]
++
+The `jsf-2.3` feature in Open Liberty makes use of link:https://myfaces.apache.org/#/core23[Apache MyFaces] JavaServer Faces implementation. With the release of Apache MyFaces 2.3.9 a number of improvements and bug fixes have been made, for more information visit the link:https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=10600&version=12349634[Apache MyFaces 2.3.9 changelog].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/15023[WASReqURLOidc cookie encodes the request url but doesn't decoded it upon successful redirection]
++
+When calling a protected `URL` including some query params, the server uses OIDC. The `URL` in the `WASReqURL` cookie was incorrectly encoded instead of the URL query params to be decoded upon successful redirection. To fix this issue instead of encoding the query string in the original request `URL`, we are leaving it as-is. That should ensure we ultimately use the `real` value of the query string. During the creation of the `WASReqURLOidc` cookie, the cookie value is encoded anyway to mitigate against malformed and potentially malicious cookie values. The `OidcRedirectServlet` code is also updated to use slightly different code to read the cookie so that the respective decoding is done.
```

Comment:
```suggestion
When calling an OIDC-protected URL that includes query parameters, the URL in the `WASReqURLOidc` cookie was incorrectly encoded. This would cause the URL query parameters to not be decoded correctly upon successful redirection to the original request URL. To fix this issue, instead of encoding the query string in the original request URL, we are leaving it as-is. That should ensure we ultimately use the correct original value of the query string. During the creation of the `WASReqURLOidc` cookie, the cookie value is encoded to mitigate against malformed and potentially malicious cookie values anyway. The `OidcRedirectServlet` class is also updated to use slightly different code to read the cookie so that the appropriate respective decoding is done.
```

Reply:
Done